Source/AssetRipper.Processing/AnimationClips/TosBuilder.cs

```diff
@@ -33,29 +33,49 @@ internal static class TosBuilder
 
 	public static IReadOnlyDictionary<uint, string> FindTOS(this IAnimationClip clip, AnimationCache cache)
 	{
+		//
+		// We don't stop adding
+		// after a match is found.
+		//
+		// This may be more performance intensive.
+		// However, it ensures a more precise
+		// recovery of all TOS data.
+		//
+		// Sometimes TOS data that is needed isnt
+		// in the specified avatar, animator, or animation even
+		// if it contains the specified animationclip.
+		//
+		// By adding all of them, we reduce
+		// the potential of missing anything.
+		//
```

Comment:
This is o(n^2). It is not just a little more performance intensive. It is quadratically more performance intensive.

If these unrelated TOS dictionaries are necessary for full recovery, I feel like the only realistic solution would be to create a global TOS dictionary and use that for everything. In the unlikely event of conflicts, continuing on is okay because Unity string equality is evaluated only by crc32 hash, which is how some developers obfuscate their strings.